athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/request/FederationRequest.java

```diff
@@ -45,7 +46,8 @@
         @JsonSubTypes.Type(value = GetSplitsRequest.class, name = "GetSplitsRequest"),
         @JsonSubTypes.Type(value = ReadRecordsRequest.class, name = "ReadRecordsRequest"),
         @JsonSubTypes.Type(value = UserDefinedFunctionRequest.class, name = "UserDefinedFunctionRequest"),
-        @JsonSubTypes.Type(value = PingRequest.class, name = "PingRequest")
+        @JsonSubTypes.Type(value = PingRequest.class, name = "PingRequest"),
+        @JsonSubTypes.Type(value = GetDataSourceCapabilitiesRequest.class, name = "GetDataSourceCapabilitiesRequest")
```

Comment:
The issue with this is that what you're suggesting only works for pushing down filters. The entire point of the framework was to avoid having to create additional remote calls per optimization, and having to make new API's for every optimization. 

I had already thought about this approach and discarded it early on because it was not extensible to all pushdown types.

Reply:
Sometimes a problem fundamentally requires extra APIs, its not always possible to have fewer APIs if the problem demands it.

In this situation, just returning a coarse grained set of capabilities is not the solution for avoiding this problem because it introduces an additional problem now where there is actually a finer granularity (per filter) on what is actually supported and whats not.

Reply:
And this problem doesn't demand it. Yes, there are edge cases for certain sources where we cannot guarantee that a filter can be entirely pushed down and thats ok. If a connector can't guarantee an optimization, then it simply doesn't advertise it.

Reply:
> If a connector can't guarantee an optimization, then it simply doesn't advertise it.

You mean in the DDB case where it can't guarantee every possible predicate pushdown, so we just assume it doesn't work at all?

If its "simply" ok to do this, then it implies that there's no impact either way, which begs the question why we would even need to advertise the capabilities at all to begin with in that case?

Reply:
That's my point though, that we do want to be able to remove portions from the query plan whenever we can. By having a coarse grained API for capabilities, you will end having an equivalently coarse grained removal or non removal of these predicates from the query plan.

Reply:
and for the vast majority of use cases, that is fine. We don't need to worry about removing portions of the query plan for sources like DDB because it has very little impact, if any. 

The main tradeoff here is number of remote calls vs fine grained pushdowns. The cases for fine grained pushdowns benefits a small percentage of queries, while the number of remote calls impacts all queries.

Reply:
> while the number of remote calls impacts all queries.

While you have good instinct here to avoid chatty protocols, keep in mind that this is just for the metadata portion of a query, it should not have a high impact on the overall query runtime.

But ok, I've already laid out my thoughts and if this is the tradeoff that you're consciously making, that's fine.